File: app/src/main/java/com/attendance/contract/MainContract.java
package com.attendance.contract;

import com.attendance.BasePresenter;
import com.attendance.BaseView;

/**
 * Created by peiqin on 3/4/2017.
 */

public interface MainContract {

    interface View extends BaseView<Presenter> {

        void showTip(String tip);

        void startRefresh();

        void stopRefresh();

        void getCourseSuccess();

    }

    interface Presenter extends BasePresenter {

        void getCourseList();

    }

}

File: app/src/main/kotlin/com.attendance/contract/MainContract.kt
package com.attendance.contract

import com.attendance.BasePresenter
import com.attendance.BaseView

/**
 * Created by peiqin on 3/4/2017.
 */

interface MainContract {

    interface View : BaseView<Presenter> {

        fun showTip(tip: String)

        fun startRefresh()

        fun stopRefresh()

        fun getCourseSuccess()

    }

    interface Presenter : BasePresenter {

        fun getCourseList()

    }

}
